### ISSUE TITLE
hevm: unexpected runtime error during symbolic execution

### ISSUE THREAD

**Issue by xwvvvvwx** [dapphub/dapptools#457]
`hevm` fails with `hevm: Maybe.fromJust: Nothing` when stepping over the `revert(0,x)` in the following yul program:

```
{
        calldatacopy(0,0,1024)

        for {} calldatasize() { mstore(1, 2) } {
                let x := 7
                mstore(4, 5)
                revert(0, x)
        }

}
```

Note that this error is not reproducible when executing concretely:

```
$hevm exec --gas 99999999999 --code 6104006000600037361560195760076005600452806000fd505b --debug --calldata $(seth --to-bytes32 $(seth --to-hex 9)
```

Or when symbolicaly executing into a simple `revert(0,7`:

```
hevm symbolic --code $(tmp=$(mktemp) && echo '{ revert(0,7) }' > $tmp && solc  --yul --yul-dialect evm $tmp |  awk 'f{print;f=0} /Binary representation:/{f=1}') --debug
```

*Steps To Reproduce*

1. `hevm symbolic --code 6104006000600037361560195760076005600452806000fd505b`

**Comment by MrChico** [comment]
I think we are currently not accepting symbolic bytes to be returned in reverts. 
I took the easy route of forcing concreteness since I expected the revertdata to be concrete error strings anyway. It's possible that it could be easy to generalize though

**Comment by xwvvvvwx** [comment]
hmmmmm, it's a little curious that we get a `Maybe.fromJust: Nothing` with symbolic args to `revert` since I see `hevm: Unexpected symbolic argument at opcode: OpRevert. Not supported (yet!)` when I run:

```
hevm equivalence --code-a 61040060006000376055565b6000600083601f8401121515601e578182fd5b8235905067ffffffffffffffff8111156035578182fd5b602083019150836020828501011115604d5760006000fd5b5b9250929050565b6061600151600051600b565b8082525050 --code-b 61040060006000376055565b6000600083601f8401121515601e578182fd5b8235905067ffffffffffffffff8111156035578182fd5b602083019150836020828501011115604d5760006000fd5b5b9250929050565b6061600151600051600b565b8082525050
```

**Comment by MrChico** [comment]
Yes. It has to do with graceful failures, done in the continuation passing style of [`forceConcrete*`](https://github.com/dapphub/dapptools/blob/master/src/hevm/src/EVM.hs#L1701) family, versus the cruder [`forceLit`](https://github.com/dapphub/dapptools/blob/master/src/hevm/src/EVM/Symbolic.hs#L49) family. The former should always be preferred, but is sometimes a little trickier to work with. I have addressed the case of `REVERT` in https://github.com/dapphub/dapptools/commit/c016fbe180ee98c9ce0d8c06c4ae3c49a87ecdc3

**Comment by MrChico** [comment]
Actually, there are two issues at play here. The first being the `fromJust` issue, which I am referring to above. The second is the fact that the equivalence error message is wrong, due to a typo which is corrected in 6a3f29ff
